src/manifold/src/boolean_result.cpp

```diff
@@ -614,9 +614,15 @@ Manifold::Impl Boolean3::Result(OpType op) const {
 
   ASSERT((expandP_ > 0) == (op == OpType::Add), logicErr,
          "Result op type not compatible with constructor op type.");
-  const int c1 = op == OpType::Intersect ? 0 : 1;
-  const int c2 = op == OpType::Add ? 1 : 0;
-  const int c3 = op == OpType::Intersect ? 1 : -1;
+
+                     // Add, Subtract, Intersect, KeepP, KeepQ
+  const int c1tab[] = { 1, 1, 0, 1, 0 };
+  const int c2tab[] = { 1, 0, 0, 0, 1 };
+  const int c3tab[] = { -1, -1, 1, 0, 0 };
```

Comment:
Let's move this [convo](https://github.com/elalish/manifold/discussions/573#discussioncomment-7283806) to these threads. What do you mean by cA and cB? But generally I think you're thinking about this right.

Reply:
I was trying to use the notation from the dissertation, p. 68.

here cA==c1, cB==c2, cI==c3 (I think).

Reply:
X12 is the new verts from intersections of edges of A with faces of B - X21 is faces of A with edges of B. X03 is just the retained original verts from A, while X30 is those from B. w03/w30 is the winding number of those verts with respect to the other manifold. 

Let's add just enough to make this operation work. Please stop asking for non-manifold output - that's a hard no, as it'll break every assumption in the library. It's a feature (the primary feature), not a bug.

Reply:
I wasn't seriously asking for non-manifold output here.  Sorry to have come across as not getting the message...

I was thinking here in terms of a temporary hack that could produce the figures I described.  And if we came up with other reasonable outputs in the future, all possible combinations could be defined in the six constants and we wouldn't have to go through the further code changes again.

What I don't understand about X12/X21/X30/X03 is how the duplicated locations are resolved.

If X30 contains _all_ the faces from the original A, and X12 includes split faces of A around the intersections, how does the simple combination operator figure out what to keep?

It seems that X30 may only contain the faces of A that are not involved in any sort of intersection.  In that case, if we visualized it, we would see a gap in X30 about one triangle wide that goes around the intersection curve.

We would then see that X12/X21 covers that gap, but with split triangles -- on each side.

But that doesn't handle the faces of A that are entirely within B, but not near the intersection curve.

So then X30 should contain only faces from original A that are entirely outside of B.

But this would require X12 to include not only the split triangles of A, but also any faces of A buried entirely in B.

For me, a figure that illustrates this would make it immediately clear.

Reply:
A diagram would be great, but I don't have a good way to draw one at the moment. The fundamental misunderstanding is you're thinking of these X values as pertaining to faces - they don't, they pertain to vertices. The included verts are used to build up edges, and then faces as part of the assembly. So, the four X sets we're talking about are entirely disjoint. All operations must include all of X12 and X21 (though potentially with reversed sign/direction). The only difference is which portions of X03 and X30 to include, which for a Boolean is by winding number, but for stamp is not.